### PR TITLE
Move flutter_assets to App.framework

### LINKED ISSUE #23839: `flutter_assets` should be embedded in App.framework
Currently, flutter_assets gets embedded directly into `Runner.app`.

For add to app, this creates additional work to embed resources into the target host app. 

It would be better if we could just embed `flutter_assets` into `App.framework` instead.

/cc @chinmaygarde @cbracken @xster @matthew-carroll

### PR DESCRIPTION
**Pull request by dnfield**
This moves the `flutter_assets` folder to `App.framework`.  The engine side is already set up to use this correctly, and has been for some time.

- Updates `xcode_backend.sh` to write `flutter_assets` to `App.framework` folder
- Removes need to copy that folder when embedding assets for Add2App
- Changes `upgradePbxProjWithFlutterAssets` to remove references to `flutter_assets` and `app.flx`.
  - Adds test for that method
  - Makes that method more efficient rather than doing multiple iterations of the pbxproj lines.  Even if we can't land this patch, we should land a fix for that.
- Updates pbxproj.tmpl files to remove references to `flutter_assets`.

This will make our Add2App story take one less step for iOS.  It will also improve our ability to support Carthage (we can just vend the frameworks and the assets are included).

Fixes #23839 

Helps with https://github.com/flutter/flutter/issues/26099

### CODE REVIEW

**Comment by dnfield** [comment]
For reviewers -  https://github.com/flutter/flutter/pull/26630/commits/1215156916cd5f88fef35ac8aff5654f000dbf6e is the meat of this change, the second commit is updating all the pbxproj files in our various samples/tests so we don't end up rewriting them every time.

**Review by chinmaygarde — APPROVED** [review]

**Comment by dnfield** [comment]
Helps with https://github.com/flutter/flutter/issues/24176

**Comment by xster** [comment]
LGTM

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -719,58 +719,19 @@ Future<bool> upgradePbxProjWithFlutterAssets(IosProject project) async {
...
-        'replaced with flutter_assets.');
-    lines.remove(l9);
-    lines.remove(l10);
-    lines.remove(l11);
-    lines.remove(l12);
+  for(final String line in lines) {
+    final Match match = oldAssets.firstMatch(line);
+    if (match != null) {
+      if (printedStatuses.add(match.group(1)))
+        printStatus('Removing obsolete reference to ${match.group(1)} from ${project.hostAppBundleName}');
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter_tools/lib/src/ios/mac.dart]: The continue statement on the next line skips adding it to the `StringBuffer`.

[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Think that'd be preferable as an `else` clause?

**Comment by dgurudot** [comment]
Causing the following error now and blank screen

Failed to find assets path for "Frameworks/App.framework/flutter_assets"

**Comment by dgurudot** [comment]
causing following issue ... plz share workaround or way forward

https://github.com/flutter/flutter/issues/33121

**Comment by dgurudot** [comment]
@chinmaygarde ... plz see the issue now.

**Comment by dnfield** [comment]
@dgurudot - this was merged over a year ago now and is being used successfully for add to app and regular uses.  It sounds like you have a new issue. Please follow up on the issue you linked, or if yours is different, file a new one with full steps to reproduce.

**Pull request merged**